context.js

```diff
@@ -37,6 +37,15 @@ module.exports = {
 		// Call the error callback and fail
 		return Promise.resolve()
 			.then(() => app.errorCb(err))
-			.then(() => context.fail(msg));
+			.then(() => context.fail(msg))
+			.catch(error => {
+				if (error) {
+					context.fail(error);
+
+					return;
+				}
+
+				context.fail(msg);
```

Comment:
```suggestion
			.catch(error => context.fail(error || msg));
```

I'm even wondering if we should do `|| msg`. But it is possible to return `Promise.reject()` in the `onError` without an error object so yeah, the fallback is necessary.